examples/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+
+foreach(example
+    4bit_palette
+    bmp2png
+    decode
+    encode
+    gzip
+    #opengl
+    optimize_png
+    png2bmp
+    png_info
+    reencode
+    #sdl
+  )
+  add_executable(example_${example}
+    example_${example}.cpp)
+  target_link_libraries(example_${example} PRIVATE lodepng)
+endforeach()
+
+# SDL2 Dependency
+find_package(SDL2 CONFIG REQUIRED)
+if (TARGET SDL2::SDL2)
+  message(STATUS "using TARGET SDL2::SDL2")
+else()
+  message(STATUS "no TARGET SDL2::SDL2, or SDL2, using variables")
```

Comment:
How can you get here? Did you expect an older version of this library without `SDL2::SDL2`? If so, maybe it should be better to create your own `SDL2Find.cmake` instead of using `SDL2_INCLUDE_DIRS` directly.